device/sycl/src/clusterization/clusterization_algorithm.sycl

```diff
@@ -392,25 +441,40 @@ clusterization_algorithm::output_type clusterization_algorithm::operator()(
         .memset(num_measurements_device.get(), 0, sizeof(unsigned int))
         .wait_and_throw();
 
-    const unsigned short max_cells_per_partition = m_max_cells_per_partition;
+    const unsigned short max_cells_per_partition =
+        m_target_cells_per_partition *
+        (MAX_CELLS_PER_THREAD / (float)TARGET_CELLS_PER_THREAD);
     const unsigned int threads_per_partition =
-        (m_max_cells_per_partition + CELLS_PER_THREAD - 1) / CELLS_PER_THREAD;
+        (m_target_cells_per_partition + TARGET_CELLS_PER_THREAD - 1) /
+        TARGET_CELLS_PER_THREAD;
+    const unsigned int num_partitions =
+        (num_cells + m_target_cells_per_partition - 1) /
+        m_target_cells_per_partition;
+    const unsigned int target_cells_per_partition =
+        m_target_cells_per_partition;
+
     ::sycl::nd_range ndrange(
         ::sycl::range<1>(num_partitions * threads_per_partition),
         ::sycl::range<1>(threads_per_partition));
 
     // Check if device is capable of allocating sufficient local memory
     assert(sizeof(index_t) * 2 * max_cells_per_partition +
-               sizeof(unsigned int) <
+               3 * sizeof(unsigned int) <
```

Comment:
Why do we need this factor three here? I don't think we multiply by three for the CUDA code, do we?

Reply:
in the CUDA code we only dynamically allocate the 2 `index_t `vectors. Here in SYCL we also need to have the 3 extra shared `unsigned int` variables (as all shared variables) allocated like this. https://github.com/guilhermeAlmeida1/traccc/blob/99ea08e1a68600226531fceb648d611c7a915429/device/sycl/src/clusterization/clusterization_algorithm.sycl#L472-L488

Now that I look at this I could've made the whole thing more compact by just allocating a 3-sized vector of unsigned's instead of 3 separate ones, so let me do that in this PR as well.

Reply:
Ok, I trust your judgement on this.